Libplanet.Tests/Action/ActionEvaluatorTest.cs

```diff
@@ -258,6 +258,31 @@ DumbAction MakeAction(Address address, char identifier, Address? transferTo = nu
                         : null);
             }
 
+            var totalSupplyGetterGetter = new Func<
+                IImmutableDictionary<Currency, FungibleAssetValue>,
+                TotalSupplyGetter
+            >(
+                totalSupplies =>
+                    currency =>
+                    {
+                        if (!currency.TotalSupplyTrackable)
+                        {
+                            var msg =
+                                $"The total supply value of the currency {currency} is not"
+                                + " trackable because it is a legacy untracked currency which might"
+                                + " have been established before the introduction of total supply"
+                                + " tracking support.";
+                            throw new TotalSupplyNotTrackableException(currency, msg);
+                        }
+
+                        if (totalSupplies.TryGetValue(currency, out FungibleAssetValue v))
+                        {
+                            return v;
+                        }
+
+                        return currency * 0;
```

Comment:
It's trivial, but this can be shorten:

```suggestion
                        return totalSupplies.TryGetValue(currency, out FungibleAssetValue v)
                            ? v
                            : currency * 0;
```